internal/cmd/load-balancer/observability-credentials/cleanup/cleanup.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/stackitcloud/stackit-cli/internal/pkg/projectname"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/load-balancer/client"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/load-balancer/utils"
+
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-sdk-go/services/loadbalancer"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "cleanup",
+		Short: "Deletes observability credentials unused by any Load Balancer",
+		Long:  "Deletes observability credentials unused by any Load Balancer.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Delete observability credentials unused by any Load Balancer`,
+				"$ stackit load-balancer observability-credentials cleanup"),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			projectLabel, err := projectname.GetProjectName(ctx, p, cmd)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get project name: %v", err)
+				projectLabel = model.ProjectId
+			}
+
+			listReq := buildListCredentialsRequest(ctx, model, apiClient)
+			resp, err := listReq.Execute()
+			if err != nil {
+				return fmt.Errorf("list Load Balancer observability credentials: %w", err)
+			}
+
+			var credentials []loadbalancer.CredentialsResponse
+			if resp.Credentials != nil && len(*resp.Credentials) > 0 {
+				credentials, err = utils.FilterCredentials(ctx, apiClient, *resp.Credentials, model.ProjectId, utils.OP_FILTER_UNUSED)
+				if err != nil {
+					return fmt.Errorf("filter Load Balancer observability credentials: %w", err)
+				}
+			}
+
+			if len(credentials) == 0 {
+				p.Info("No unused observability credentials found on project %q\n", projectLabel)
+				return nil
+			}
+
+			if !model.AssumeYes {
+				prompt := "Will delete the following unused observability credentials: \n"
+				for _, credential := range credentials {
+					if credential.DisplayName == nil || credential.Username == nil {
+						return fmt.Errorf("delete Load Balancer observability credentials: missing display name or username")
+					}
+					name := *credential.DisplayName
+					username := *credential.Username
+					prompt += fmt.Sprintf("  - %s (username: %q)\n", name, username)
+				}
+				prompt += fmt.Sprintf("Are you sure you want to delete unused observability credentials on project %q? (This cannot be undone)", projectLabel)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			for _, credential := range credentials {
+				if credential.CredentialsRef == nil {
+					return fmt.Errorf("delete Load Balancer observability credentials: missing credentials reference")
+				}
```

Comment:
I think this will never happen

Reply:
Should I not check for it then? I think it makes sense to avoid a segmentation fault being displayed to the user

Reply:
Yeah it's fine, there are a lot of places where this could happen theoretically but the object is required in the API response